include/uapi/linux/pkt_sched.h

```diff
@@ -124,6 +124,21 @@ struct tc_fifo_qopt {
 	__u32	limit;	/* Queue length: bytes for bfifo, packets for pfifo */
 };
 
+/* DSPRIO section */
+
+#define DSPRIO_MAX_PRIORITY 64
+
+struct tc_dsprio_qopt {
+	__u32	limit; 	    	/* Queue length in packets. */
+	__u16	noip_dfltp; 	/* Default priority for non-IP packets. */
+
+	/* Stats. */
+	__u32 qlen[DSPRIO_MAX_PRIORITY - 1]; /* Individual dsprio queue
```

Comment:
We need to drop the array `qlen` here and follow the example of `net/sched/sch_fq_codel.c` as we've agreed with Cong: https://www.spinics.net/lists/netdev/msg501328.html

Essentially, the steps are as follows:
1. Add `static const struct Qdisc_class_ops dsprio_class_ops` just before `struct Qdisc_ops dsprio_qdisc_ops`;
2. Add field `.cl_ops = &dsprio_class_ops,` to `struct Qdisc_ops dsprio_qdisc_ops`;
2. Define the following fields/methods of `dsprio_class_ops` as it's done in `net/sched/sch_api.c`:  `.find`, `.walk`, `.leaf`, `.dump`, `.dump_stats`. The method `dsprio_dump_class_stats()` will be the one that will send the number of packets in each priority.

Once this change is in place, write userspace code (i.e. extend the command `ip`) to test that this new feature is working properly.

Reply:
Once you finish removing the array `qlen`, we'll no longer need the fields `highest_prio` and `lowest_prio` of `struct tc_dsprio_qopt` since an application can easily infer them from the number of packets in each priority.